builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -222,13 +227,19 @@ func (r *ResourceProvisioner) decodeConfig(c *terraform.ResourceConfig) (*Provis
 	}
 
 	if p.ValidationKeyPath != "" {
-		keyPath, err := homedir.Expand(p.ValidationKeyPath)
+		vkeyPath, err := homedir.Expand(p.ValidationKeyPath)
 		if err != nil {
 			return nil, fmt.Errorf("Error expanding the validation key path: %v", err)
 		}
-		p.ValidationKeyPath = keyPath
-	}
-
+		p.ValidationKeyPath = vkeyPath
+	}
+        if p.SecretKeyPath != "" {
+                skeyPath, err := homedir.Expand(p.SecretKeyPath)
```

Comment:
Not sure if I understand your comment. My point here was about the `keyPath` variable being within separated scopes, so that you did not have to think of separate names for the variable in their own scopes.

It's just a nit and I noticed you updated it in your latest commits, so looks fine now :smiley: